README.md

```diff
@@ -231,6 +280,38 @@ lspconfig.SERVER.setup{config}
     `new_config.cmd = cmd` is present within the function body.
 ```
 
-# Individual server documentation
+## Debugging
+The two most common points of failure are 
 
-See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
+1. The language server is not installed. You should be able to run the `cmd` 
+defined in each lua module from the command line.
+
+2. Not triggering root detection. The language server will only start if it
+is opened in a directory, or child directory, containing a file which signals
+the *root* of the project. Most of the time, this is a `.git` folder, but each server
+defines the root config in the lua file.
+
+Before reporting a bug, check your logs. Add the following to your init.vim
+to enable logging 
+
+```vim
+lua << EOF
+vim.lsp.set_log_level("debug")
+EOF
+```
+Attempt to run the language server, and view the log present at:
+```
+:lua print(vim.lsp.get_log_path())
+```
+Most of the time, the reason for failure is present in the logs.
+
+## Contributions
+
```

Comment:
Maybe add explicitly "If you are missing a language server on the list [CONFIG.md], contributing a new configuration for it would be appreciated. You can follow these steps:"

Reply:
Done!